lib/spack/spack/build_systems/autotools.py

```diff
@@ -435,7 +435,7 @@ def _do_patch_libtool(self):
                 "crtbeginS.o",
                 "crtendS.o",
```

Comment:
This and above is now passed as a regular expression, therefore I would escape the dots.
```suggestion
                r"crtendS\.o",
```